src/managers/CursorManager.cpp

```diff
@@ -396,63 +399,141 @@ void CCursorManager::SXCursorManager::loadTheme(const std::string& name, int siz
+            }
         }
-    }
 
-    defaultCursor          = makeShared<SXCursor>();
-    defaultCursor->size    = {(int)img->width, (int)img->height};
-    defaultCursor->hotspot = {(int)img->xhot, (int)img->yhot};
+        auto xcursor     = makeShared<SXCursor>();
+        xcursor->size    = {(int)xImage->width, (int)xImage->height};
+        xcursor->hotspot = {(int)xImage->xhot, (int)xImage->yhot};
 
-    defaultCursor->pixels.resize(img->width * img->height);
-    std::memcpy(defaultCursor->pixels.data(), img->pixels, img->width * img->height * sizeof(uint32_t));
+        xcursor->pixels.resize(xImage->width * xImage->height);
+        std::memcpy(xcursor->pixels.data(), xImage->pixels, xImage->width * xImage->height * sizeof(uint32_t));
 
-    themeLoaded = true;
+        cursors.emplace_back(std::string{shape}, xcursor);
 
-    XcursorImageDestroy(img);
+        XcursorImageDestroy(xImage);
+    }
 
-    // gather as many shapes as we can find.
-    cursors.clear();
+    if (cursors.empty()) {
+        Debug::log(ERR, "XCursor failed finding any shapes in theme \"{}\".", themeName);
+        return;
+    }
 
-    for (auto& shape : CURSOR_SHAPE_NAMES) {
-        int id = -1;
-        for (size_t i = 0; i < XCURSOR_STANDARD_NAMES.size(); ++i) {
-            if (XCURSOR_STANDARD_NAMES.at(i) == std::string{shape}) {
-                id = i;
-                break;
-            }
-        }
+    themeLoaded = true;
 
-        if (id < 0) {
-            Debug::log(LOG, "XCursor has no shape {}, skipping", shape);
+    for (auto const& shape : CURSOR_SHAPE_NAMES) {
+        auto legacyName = getLegacyShapeName(shape);
+        if (legacyName.empty())
             continue;
-        }
 
-        auto xImage = XcursorShapeLoadImage(id << 1 /* wtf xcursor? */, themeName.c_str(), size);
+        auto it = std::find_if(cursors.begin(), cursors.end(), [&legacyName](auto const& c) { return c.first == legacyName; });
 
-        if (!xImage) {
-            Debug::log(LOG, "XCursor failed to find a shape with name {}, skipping", shape);
+        if (it == cursors.end()) {
+            Debug::log(LOG, "XCursor failed to find a legacy shape with name {}, skipping", legacyName);
             continue;
         }
 
-        auto xcursor     = makeShared<SXCursor>();
-        xcursor->size    = {(int)xImage->width, (int)xImage->height};
-        xcursor->hotspot = {(int)xImage->xhot, (int)xImage->yhot};
+        cursors.emplace_back(shape, it->second);
+    }
 
-        xcursor->pixels.resize(xImage->width * xImage->height);
-        std::memcpy(xcursor->pixels.data(), xImage->pixels, xImage->width * xImage->height * sizeof(uint32_t));
+    // set default cursor
+    for (auto const& c : cursors) {
+        if (c.first == "left_ptr" || c.first == "arrow") {
+            defaultCursor = c.second;
+            break;
+        }
+    }
 
-        cursors.emplace(std::string{shape}, xcursor);
+    // just assign the first one then.
+    if (!defaultCursor)
```

Comment:
you didnt reset default cursor this will fail if you change theme cuz there is a default cursor already boyyyyyyyy

Reply:
ah truetrue

Reply:
hahahhasheheheheheheheehheehhyeheheehehehrewyuhrteytfhreytgheryutgberybgertugbreugberbnviuerfb giuerbguieriugerhiugberiugberuigbn uierngiuoerni ug er9ui ger

Reply:
The provided string appears to be a random sequence of characters without an obvious encoding pattern. To determine if it contains any hidden message or meaningful content, I can attempt various decoding techniques. However, if it is genuinely random, it may not yield any sensible results.

Let's start by checking for common encoding schemes such as Base64, hexadecimal, or simple substitution ciphers.

Could you provide any context or hints about how this message might be encoded? That would help in narrowing down the decoding method.

Reply:
ok changed, im now resetting defaultcursor too

Reply:
should i account for user errors, like setting the xcursor theme to a non existing one or should it simply turn into an empty theme

Reply:
not that im entirerly sure on how to do that but

Reply:
why not just be a funny funny trolle XD memes and make it a 32x32 hyprland icon that would be funny

Reply:
Hm can see tomorrow if i can make something like that, going to bed

Reply:
I can donate the bytes if you want

Reply:
this is how it looks now. a white H as in Hyprland! :) this should sort of fix #7056 if a user doesnt set a xcursor theme it will fallback to the white H. could use some testing around incase i missed something.

https://github.com/user-attachments/assets/c229d8bd-73ef-406f-8fba-675f41931583

Reply:
doesnt set a theme and the default one not existing. i mean

Reply:
can you please not earrape next time

Reply:
Oh didnt notice it recorded my wife on the internal mic sorry